diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/.gitignore rails 5.0/.gitignore
--- rails 4.2/.gitignore	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/.gitignore	2022-10-13 09:23:33.000000000 -0700
@@ -13,5 +13,9 @@
 
 # Ignore all logfiles and tempfiles.
 /log/*
+/tmp/*
 !/log/.keep
-/tmp
+!/tmp/.keep
+
+# Ignore Byebug command history file.
+.byebug_history
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/Gemfile rails 5.0/Gemfile
--- rails 4.2/Gemfile	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/Gemfile	2022-10-13 09:23:33.000000000 -0700
@@ -1,47 +1,53 @@
 source 'https://rubygems.org'
 
+git_source(:github) do |repo_name|
+  repo_name = "#{repo_name}/#{repo_name}" unless repo_name.include?("/")
+  "https://github.com/#{repo_name}.git"
+end
+
 
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
-gem 'rails', '4.2.11.3'
+gem 'rails', '~> 5.0.7', '>= 5.0.7.2'
 # Use sqlite3 as the database for Active Record
 gem 'sqlite3'
+# Use Puma as the app server
+gem 'puma', '~> 3.0'
 # Use SCSS for stylesheets
 gem 'sass-rails', '~> 5.0'
 # Use Uglifier as compressor for JavaScript assets
 gem 'uglifier', '>= 1.3.0'
 # Use CoffeeScript for .coffee assets and views
-gem 'coffee-rails', '~> 4.1.0'
+gem 'coffee-rails', '~> 4.2'
 # See https://github.com/rails/execjs#readme for more supported runtimes
 # gem 'therubyracer', platforms: :ruby
 
 # Use jquery as the JavaScript library
 gem 'jquery-rails'
-# Turbolinks makes following links in your web application faster. Read more: https://github.com/rails/turbolinks
-gem 'turbolinks'
+# Turbolinks makes navigating your web application faster. Read more: https://github.com/turbolinks/turbolinks
+gem 'turbolinks', '~> 5'
 # Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
-gem 'jbuilder', '~> 2.0'
-# bundle exec rake doc:rails generates the API under doc/api.
-gem 'sdoc', '~> 0.4.0', group: :doc
-
+gem 'jbuilder', '~> 2.5'
+# Use Redis adapter to run Action Cable in production
+# gem 'redis', '~> 3.0'
 # Use ActiveModel has_secure_password
 # gem 'bcrypt', '~> 3.1.7'
 
-# Use Unicorn as the app server
-# gem 'unicorn'
-
 # Use Capistrano for deployment
 # gem 'capistrano-rails', group: :development
 
 group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
-  gem 'byebug'
+  gem 'byebug', platform: :mri
 end
 
 group :development do
-  # Access an IRB console on exception pages or by using <%= console %> in views
-  gem 'web-console', '~> 2.0'
-
+  # Access an IRB console on exception pages or by using <%= console %> anywhere in the code.
+  gem 'web-console', '>= 3.3.0'
+  gem 'listen', '~> 3.0.5'
   # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
   gem 'spring'
+  gem 'spring-watcher-listen', '~> 2.0.0'
 end
 
+# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
+gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/Gemfile.lock rails 5.0/Gemfile.lock
--- rails 4.2/Gemfile.lock	2022-10-13 15:13:40.000000000 -0700
+++ rails 5.0/Gemfile.lock	2022-10-13 09:23:41.000000000 -0700
@@ -1,113 +1,118 @@
 GEM
   remote: https://rubygems.org/
   specs:
-    actionmailer (4.2.11.3)
-      actionpack (= 4.2.11.3)
-      actionview (= 4.2.11.3)
-      activejob (= 4.2.11.3)
+    actioncable (5.0.7.2)
+      actionpack (= 5.0.7.2)
+      nio4r (>= 1.2, < 3.0)
+      websocket-driver (~> 0.6.1)
+    actionmailer (5.0.7.2)
+      actionpack (= 5.0.7.2)
+      actionview (= 5.0.7.2)
+      activejob (= 5.0.7.2)
       mail (~> 2.5, >= 2.5.4)
-      rails-dom-testing (~> 1.0, >= 1.0.5)
-    actionpack (4.2.11.3)
-      actionview (= 4.2.11.3)
-      activesupport (= 4.2.11.3)
-      rack (~> 1.6)
-      rack-test (~> 0.6.2)
-      rails-dom-testing (~> 1.0, >= 1.0.5)
+      rails-dom-testing (~> 2.0)
+    actionpack (5.0.7.2)
+      actionview (= 5.0.7.2)
+      activesupport (= 5.0.7.2)
+      rack (~> 2.0)
+      rack-test (~> 0.6.3)
+      rails-dom-testing (~> 2.0)
       rails-html-sanitizer (~> 1.0, >= 1.0.2)
-    actionview (4.2.11.3)
-      activesupport (= 4.2.11.3)
+    actionview (5.0.7.2)
+      activesupport (= 5.0.7.2)
       builder (~> 3.1)
       erubis (~> 2.7.0)
-      rails-dom-testing (~> 1.0, >= 1.0.5)
+      rails-dom-testing (~> 2.0)
       rails-html-sanitizer (~> 1.0, >= 1.0.3)
-    activejob (4.2.11.3)
-      activesupport (= 4.2.11.3)
-      globalid (>= 0.3.0)
-    activemodel (4.2.11.3)
-      activesupport (= 4.2.11.3)
-      builder (~> 3.1)
-    activerecord (4.2.11.3)
-      activemodel (= 4.2.11.3)
-      activesupport (= 4.2.11.3)
-      arel (~> 6.0)
-    activesupport (4.2.11.3)
-      i18n (~> 0.7)
+    activejob (5.0.7.2)
+      activesupport (= 5.0.7.2)
+      globalid (>= 0.3.6)
+    activemodel (5.0.7.2)
+      activesupport (= 5.0.7.2)
+    activerecord (5.0.7.2)
+      activemodel (= 5.0.7.2)
+      activesupport (= 5.0.7.2)
+      arel (~> 7.0)
+    activesupport (5.0.7.2)
+      concurrent-ruby (~> 1.0, >= 1.0.2)
+      i18n (>= 0.7, < 2)
       minitest (~> 5.1)
-      thread_safe (~> 0.3, >= 0.3.4)
       tzinfo (~> 1.1)
-    arel (6.0.4)
-    binding_of_caller (1.0.0)
-      debug_inspector (>= 0.0.1)
+    arel (7.1.4)
+    bindex (0.8.1)
     builder (3.2.4)
     byebug (11.1.3)
-    coffee-rails (4.1.1)
+    coffee-rails (4.2.2)
       coffee-script (>= 2.2.0)
-      railties (>= 4.0.0, < 5.1.x)
+      railties (>= 4.0.0)
     coffee-script (2.4.1)
       coffee-script-source
       execjs
     coffee-script-source (1.12.2)
     concurrent-ruby (1.1.10)
     crass (1.0.6)
-    debug_inspector (1.1.0)
     erubis (2.7.0)
     execjs (2.8.1)
     ffi (1.15.5)
-    globalid (0.4.2)
-      activesupport (>= 4.2.0)
-    i18n (0.9.5)
+    globalid (1.0.0)
+      activesupport (>= 5.0)
+    i18n (1.12.0)
       concurrent-ruby (~> 1.0)
-    jbuilder (2.9.1)
-      activesupport (>= 4.2.0)
+    jbuilder (2.11.5)
+      actionview (>= 5.0.0)
+      activesupport (>= 5.0.0)
     jquery-rails (4.5.0)
       rails-dom-testing (>= 1, < 3)
       railties (>= 4.2.0)
       thor (>= 0.14, < 2.0)
-    json (1.8.6)
+    listen (3.0.8)
+      rb-fsevent (~> 0.9, >= 0.9.4)
+      rb-inotify (~> 0.9, >= 0.9.7)
     loofah (2.19.0)
       crass (~> 1.0.2)
       nokogiri (>= 1.5.9)
     mail (2.7.1)
       mini_mime (>= 0.1.1)
+    method_source (1.0.0)
     mini_mime (1.1.2)
     mini_portile2 (2.8.0)
     minitest (5.16.3)
+    nio4r (2.5.8)
     nokogiri (1.13.8)
       mini_portile2 (~> 2.8.0)
       racc (~> 1.4)
+    puma (3.12.6)
     racc (1.6.0)
-    rack (1.6.13)
+    rack (2.2.4)
     rack-test (0.6.3)
       rack (>= 1.0)
-    rails (4.2.11.3)
-      actionmailer (= 4.2.11.3)
-      actionpack (= 4.2.11.3)
-      actionview (= 4.2.11.3)
-      activejob (= 4.2.11.3)
-      activemodel (= 4.2.11.3)
-      activerecord (= 4.2.11.3)
-      activesupport (= 4.2.11.3)
-      bundler (>= 1.3.0, < 2.0)
-      railties (= 4.2.11.3)
-      sprockets-rails
-    rails-deprecated_sanitizer (1.0.4)
-      activesupport (>= 4.2.0.alpha)
-    rails-dom-testing (1.0.9)
-      activesupport (>= 4.2.0, < 5.0)
-      nokogiri (~> 1.6)
-      rails-deprecated_sanitizer (>= 1.0.1)
+    rails (5.0.7.2)
+      actioncable (= 5.0.7.2)
+      actionmailer (= 5.0.7.2)
+      actionpack (= 5.0.7.2)
+      actionview (= 5.0.7.2)
+      activejob (= 5.0.7.2)
+      activemodel (= 5.0.7.2)
+      activerecord (= 5.0.7.2)
+      activesupport (= 5.0.7.2)
+      bundler (>= 1.3.0)
+      railties (= 5.0.7.2)
+      sprockets-rails (>= 2.0.0)
+    rails-dom-testing (2.0.3)
+      activesupport (>= 4.2.0)
+      nokogiri (>= 1.6)
     rails-html-sanitizer (1.4.3)
       loofah (~> 2.3)
-    railties (4.2.11.3)
-      actionpack (= 4.2.11.3)
-      activesupport (= 4.2.11.3)
+    railties (5.0.7.2)
+      actionpack (= 5.0.7.2)
+      activesupport (= 5.0.7.2)
+      method_source
       rake (>= 0.8.7)
       thor (>= 0.18.1, < 2.0)
     rake (13.0.6)
     rb-fsevent (0.11.2)
     rb-inotify (0.10.1)
       ffi (~> 1.0)
-    rdoc (4.3.0)
     sass (3.7.4)
       sass-listen (~> 4.0.0)
     sass-listen (4.0.0)
@@ -119,10 +124,10 @@
       sprockets (>= 2.8, < 4.0)
       sprockets-rails (>= 2.0, < 4.0)
       tilt (>= 1.1, < 3)
-    sdoc (0.4.2)
-      json (~> 1.7, >= 1.7.7)
-      rdoc (~> 4.0)
-    spring (3.1.1)
+    spring (2.1.1)
+    spring-watcher-listen (2.0.1)
+      listen (>= 2.7, < 4.0)
+      spring (>= 1.2, < 3.0)
     sprockets (3.7.2)
       concurrent-ruby (~> 1.0)
       rack (> 1, < 3)
@@ -142,28 +147,34 @@
       thread_safe (~> 0.1)
     uglifier (4.2.0)
       execjs (>= 0.3.0, < 3)
-    web-console (2.3.0)
-      activemodel (>= 4.0)
-      binding_of_caller (>= 0.7.2)
-      railties (>= 4.0)
-      sprockets-rails (>= 2.0, < 4.0)
+    web-console (3.7.0)
+      actionview (>= 5.0)
+      activemodel (>= 5.0)
+      bindex (>= 0.4.0)
+      railties (>= 5.0)
+    websocket-driver (0.6.5)
+      websocket-extensions (>= 0.1.0)
+    websocket-extensions (0.1.5)
 
 PLATFORMS
   ruby
 
 DEPENDENCIES
   byebug
-  coffee-rails (~> 4.1.0)
-  jbuilder (~> 2.0)
+  coffee-rails (~> 4.2)
+  jbuilder (~> 2.5)
   jquery-rails
-  rails (= 4.2.11.3)
+  listen (~> 3.0.5)
+  puma (~> 3.0)
+  rails (~> 5.0.7, >= 5.0.7.2)
   sass-rails (~> 5.0)
-  sdoc (~> 0.4.0)
   spring
+  spring-watcher-listen (~> 2.0.0)
   sqlite3
-  turbolinks
+  turbolinks (~> 5)
+  tzinfo-data
   uglifier (>= 1.3.0)
-  web-console (~> 2.0)
+  web-console (>= 3.3.0)
 
 BUNDLED WITH
-   1.17.2
+   2.1.4
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/README.md rails 5.0/README.md
--- rails 4.2/README.md	1969-12-31 16:00:00.000000000 -0800
+++ rails 5.0/README.md	2022-10-13 09:23:33.000000000 -0700
@@ -0,0 +1,24 @@
+# README
+
+This README would normally document whatever steps are necessary to get the
+application up and running.
+
+Things you may want to cover:
+
+* Ruby version
+
+* System dependencies
+
+* Configuration
+
+* Database creation
+
+* Database initialization
+
+* How to run the test suite
+
+* Services (job queues, cache servers, search engines, etc.)
+
+* Deployment instructions
+
+* ...
Only in rails 4.2: README.rdoc
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/Rakefile rails 5.0/Rakefile
--- rails 4.2/Rakefile	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/Rakefile	2022-10-13 09:23:33.000000000 -0700
@@ -1,6 +1,6 @@
 # Add your own tasks in files placed in lib/tasks ending in .rake,
 # for example lib/tasks/capistrano.rake, and they will automatically be available to Rake.
 
-require File.expand_path('../config/application', __FILE__)
+require_relative 'config/application'
 
 Rails.application.load_tasks
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/app/assets/config/manifest.js rails 5.0/app/assets/config/manifest.js
--- rails 4.2/app/assets/config/manifest.js	1969-12-31 16:00:00.000000000 -0800
+++ rails 5.0/app/assets/config/manifest.js	2022-10-13 09:23:33.000000000 -0700
@@ -0,0 +1,3 @@
+//= link_tree ../images
+//= link_directory ../javascripts .js
+//= link_directory ../stylesheets .css
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/app/assets/javascripts/application.js rails 5.0/app/assets/javascripts/application.js
--- rails 4.2/app/assets/javascripts/application.js	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/app/assets/javascripts/application.js	2022-10-13 09:23:33.000000000 -0700
@@ -5,7 +5,7 @@
 // or any plugin's vendor/assets/javascripts directory can be referenced here using a relative path.
 //
 // It's not advisable to add code directly here, but if you do, it'll appear at the bottom of the
-// compiled file.
+// compiled file. JavaScript code in this file should be added after the last require_* statement.
 //
 // Read Sprockets README (https://github.com/rails/sprockets#sprockets-directives) for details
 // about supported directives.
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/app/assets/javascripts/cable.js rails 5.0/app/assets/javascripts/cable.js
--- rails 4.2/app/assets/javascripts/cable.js	1969-12-31 16:00:00.000000000 -0800
+++ rails 5.0/app/assets/javascripts/cable.js	2022-10-13 09:23:33.000000000 -0700
@@ -0,0 +1,13 @@
+// Action Cable provides the framework to deal with WebSockets in Rails.
+// You can generate new channels where WebSocket features live using the rails generate channel command.
+//
+//= require action_cable
+//= require_self
+//= require_tree ./channels
+
+(function() {
+  this.App || (this.App = {});
+
+  App.cable = ActionCable.createConsumer();
+
+}).call(this);
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/app/assets/stylesheets/application.css rails 5.0/app/assets/stylesheets/application.css
--- rails 4.2/app/assets/stylesheets/application.css	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/app/assets/stylesheets/application.css	2022-10-13 09:23:33.000000000 -0700
@@ -6,9 +6,9 @@
  * or any plugin's vendor/assets/stylesheets directory can be referenced here using a relative path.
  *
  * You're free to add application-wide styles to this file and they'll appear at the bottom of the
- * compiled file so the styles you add here take precedence over styles defined in any styles
- * defined in the other CSS/SCSS files in this directory. It is generally better to create a new
- * file per style scope.
+ * compiled file so the styles you add here take precedence over styles defined in any other CSS/SCSS
+ * files in this directory. Styles in this file should be added after the last require_* statement.
+ * It is generally better to create a new file per style scope.
  *
  *= require_tree .
  *= require_self
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/app/channels/application_cable/channel.rb rails 5.0/app/channels/application_cable/channel.rb
--- rails 4.2/app/channels/application_cable/channel.rb	1969-12-31 16:00:00.000000000 -0800
+++ rails 5.0/app/channels/application_cable/channel.rb	2022-10-13 09:23:33.000000000 -0700
@@ -0,0 +1,4 @@
+module ApplicationCable
+  class Channel < ActionCable::Channel::Base
+  end
+end
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/app/channels/application_cable/connection.rb rails 5.0/app/channels/application_cable/connection.rb
--- rails 4.2/app/channels/application_cable/connection.rb	1969-12-31 16:00:00.000000000 -0800
+++ rails 5.0/app/channels/application_cable/connection.rb	2022-10-13 09:23:33.000000000 -0700
@@ -0,0 +1,4 @@
+module ApplicationCable
+  class Connection < ActionCable::Connection::Base
+  end
+end
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/app/controllers/application_controller.rb rails 5.0/app/controllers/application_controller.rb
--- rails 4.2/app/controllers/application_controller.rb	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/app/controllers/application_controller.rb	2022-10-13 09:23:33.000000000 -0700
@@ -1,5 +1,3 @@
 class ApplicationController < ActionController::Base
-  # Prevent CSRF attacks by raising an exception.
-  # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
 end
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/app/jobs/application_job.rb rails 5.0/app/jobs/application_job.rb
--- rails 4.2/app/jobs/application_job.rb	1969-12-31 16:00:00.000000000 -0800
+++ rails 5.0/app/jobs/application_job.rb	2022-10-13 09:23:33.000000000 -0700
@@ -0,0 +1,2 @@
+class ApplicationJob < ActiveJob::Base
+end
Only in rails 4.2/app/mailers: .keep
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/app/mailers/application_mailer.rb rails 5.0/app/mailers/application_mailer.rb
--- rails 4.2/app/mailers/application_mailer.rb	1969-12-31 16:00:00.000000000 -0800
+++ rails 5.0/app/mailers/application_mailer.rb	2022-10-13 09:23:33.000000000 -0700
@@ -0,0 +1,4 @@
+class ApplicationMailer < ActionMailer::Base
+  default from: 'from@example.com'
+  layout 'mailer'
+end
Only in rails 4.2/app/models: .keep
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/app/models/application_record.rb rails 5.0/app/models/application_record.rb
--- rails 4.2/app/models/application_record.rb	1969-12-31 16:00:00.000000000 -0800
+++ rails 5.0/app/models/application_record.rb	2022-10-13 09:23:33.000000000 -0700
@@ -0,0 +1,3 @@
+class ApplicationRecord < ActiveRecord::Base
+  self.abstract_class = true
+end
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/app/views/layouts/application.html.erb rails 5.0/app/views/layouts/application.html.erb
--- rails 4.2/app/views/layouts/application.html.erb	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/app/views/layouts/application.html.erb	2022-10-13 09:23:33.000000000 -0700
@@ -1,14 +1,14 @@
 <!DOCTYPE html>
 <html>
-<head>
-  <title>R42</title>
-  <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
-  <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
+  <head>
+    <title>R50</title>
   <%= csrf_meta_tags %>
-</head>
-<body>
 
-<%= yield %>
+    <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+    <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
+  </head>
 
-</body>
+  <body>
+    <%= yield %>
+  </body>
 </html>
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/app/views/layouts/mailer.html.erb rails 5.0/app/views/layouts/mailer.html.erb
--- rails 4.2/app/views/layouts/mailer.html.erb	1969-12-31 16:00:00.000000000 -0800
+++ rails 5.0/app/views/layouts/mailer.html.erb	2022-10-13 09:23:33.000000000 -0700
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+    <style>
+      /* Email styles need to be inline */
+    </style>
+  </head>
+
+  <body>
+    <%= yield %>
+  </body>
+</html>
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/app/views/layouts/mailer.text.erb rails 5.0/app/views/layouts/mailer.text.erb
--- rails 4.2/app/views/layouts/mailer.text.erb	1969-12-31 16:00:00.000000000 -0800
+++ rails 5.0/app/views/layouts/mailer.text.erb	2022-10-13 09:23:33.000000000 -0700
@@ -0,0 +1 @@
+<%= yield %>
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/bin/rails rails 5.0/bin/rails
--- rails 4.2/bin/rails	2022-10-13 15:13:41.000000000 -0700
+++ rails 5.0/bin/rails	2022-10-13 09:23:41.000000000 -0700
@@ -4,6 +4,6 @@
 rescue LoadError => e
   raise unless e.message.include?('spring')
 end
-APP_PATH = File.expand_path('../../config/application', __FILE__)
+APP_PATH = File.expand_path('../config/application', __dir__)
 require_relative '../config/boot'
 require 'rails/commands'
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/bin/setup rails 5.0/bin/setup
--- rails 4.2/bin/setup	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/bin/setup	2022-10-13 09:23:33.000000000 -0700
@@ -1,29 +1,34 @@
 #!/usr/bin/env ruby
 require 'pathname'
+require 'fileutils'
+include FileUtils
 
 # path to your application root.
 APP_ROOT = Pathname.new File.expand_path('../../',  __FILE__)
 
-Dir.chdir APP_ROOT do
+def system!(*args)
+  system(*args) || abort("\n== Command #{args} failed ==")
+end
+
+chdir APP_ROOT do
   # This script is a starting point to setup your application.
-  # Add necessary setup steps to this file:
+  # Add necessary setup steps to this file.
 
-  puts "== Installing dependencies =="
-  system "gem install bundler --conservative"
-  system "bundle check || bundle install"
+  puts '== Installing dependencies =='
+  system! 'gem install bundler --conservative'
+  system('bundle check') || system!('bundle install')
 
   # puts "\n== Copying sample files =="
-  # unless File.exist?("config/database.yml")
-  #   system "cp config/database.yml.sample config/database.yml"
+  # unless File.exist?('config/database.yml')
+  #   cp 'config/database.yml.sample', 'config/database.yml'
   # end
 
   puts "\n== Preparing database =="
-  system "bin/rake db:setup"
+  system! 'bin/rails db:setup'
 
   puts "\n== Removing old logs and tempfiles =="
-  system "rm -f log/*"
-  system "rm -rf tmp/cache"
+  system! 'bin/rails log:clear tmp:clear'
 
   puts "\n== Restarting application server =="
-  system "touch tmp/restart.txt"
+  system! 'bin/rails restart'
 end
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/bin/update rails 5.0/bin/update
--- rails 4.2/bin/update	1969-12-31 16:00:00.000000000 -0800
+++ rails 5.0/bin/update	2022-10-13 09:23:33.000000000 -0700
@@ -0,0 +1,29 @@
+#!/usr/bin/env ruby
+require 'pathname'
+require 'fileutils'
+include FileUtils
+
+# path to your application root.
+APP_ROOT = Pathname.new File.expand_path('../../', __FILE__)
+
+def system!(*args)
+  system(*args) || abort("\n== Command #{args} failed ==")
+end
+
+chdir APP_ROOT do
+  # This script is a way to update your development environment automatically.
+  # Add necessary update steps to this file.
+
+  puts '== Installing dependencies =='
+  system! 'gem install bundler --conservative'
+  system('bundle check') || system!('bundle install')
+
+  puts "\n== Updating database =="
+  system! 'bin/rails db:migrate'
+
+  puts "\n== Removing old logs and tempfiles =="
+  system! 'bin/rails log:clear tmp:clear'
+
+  puts "\n== Restarting application server =="
+  system! 'bin/rails restart'
+end
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/config/application.rb rails 5.0/config/application.rb
--- rails 4.2/config/application.rb	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/config/application.rb	2022-10-13 09:23:33.000000000 -0700
@@ -1,4 +1,4 @@
-require File.expand_path('../boot', __FILE__)
+require_relative 'boot'
 
 require 'rails/all'
 
@@ -6,21 +6,10 @@
 # you've limited to :test, :development, or :production.
 Bundler.require(*Rails.groups)
 
-module R42
+module R50
   class Application < Rails::Application
     # Settings in config/environments/* take precedence over those specified here.
     # Application configuration should go into files in config/initializers
     # -- all .rb files in that directory are automatically loaded.
-
-    # Set Time.zone default to the specified zone and make Active Record auto-convert to this zone.
-    # Run "rake -D time" for a list of tasks for finding time zone names. Default is UTC.
-    # config.time_zone = 'Central Time (US & Canada)'
-
-    # The default locale is :en and all translations from config/locales/*.rb,yml are auto loaded.
-    # config.i18n.load_path += Dir[Rails.root.join('my', 'locales', '*.{rb,yml}').to_s]
-    # config.i18n.default_locale = :de
-
-    # Do not swallow errors in after_commit/after_rollback callbacks.
-    config.active_record.raise_in_transactional_callbacks = true
   end
 end
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/config/boot.rb rails 5.0/config/boot.rb
--- rails 4.2/config/boot.rb	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/config/boot.rb	2022-10-13 09:23:33.000000000 -0700
@@ -1,3 +1,3 @@
-ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../../Gemfile', __FILE__)
+ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../Gemfile', __dir__)
 
 require 'bundler/setup' # Set up gems listed in the Gemfile.
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/config/cable.yml rails 5.0/config/cable.yml
--- rails 4.2/config/cable.yml	1969-12-31 16:00:00.000000000 -0800
+++ rails 5.0/config/cable.yml	2022-10-13 09:23:33.000000000 -0700
@@ -0,0 +1,9 @@
+development:
+  adapter: async
+
+test:
+  adapter: async
+
+production:
+  adapter: redis
+  url: redis://localhost:6379/1
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/config/environment.rb rails 5.0/config/environment.rb
--- rails 4.2/config/environment.rb	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/config/environment.rb	2022-10-13 09:23:33.000000000 -0700
@@ -1,5 +1,5 @@
 # Load the Rails application.
-require File.expand_path('../application', __FILE__)
+require_relative 'application'
 
 # Initialize the Rails application.
 Rails.application.initialize!
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/config/environments/development.rb rails 5.0/config/environments/development.rb
--- rails 4.2/config/environments/development.rb	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/config/environments/development.rb	2022-10-13 09:23:33.000000000 -0700
@@ -9,13 +9,28 @@
   # Do not eager load code on boot.
   config.eager_load = false
 
-  # Show full error reports and disable caching.
+  # Show full error reports.
   config.consider_all_requests_local       = true
+
+  # Enable/disable caching. By default caching is disabled.
+  if Rails.root.join('tmp/caching-dev.txt').exist?
+    config.action_controller.perform_caching = true
+
+    config.cache_store = :memory_store
+    config.public_file_server.headers = {
+      'Cache-Control' => 'public, max-age=172800'
+    }
+  else
   config.action_controller.perform_caching = false
 
+    config.cache_store = :null_store
+  end
+
   # Don't care if the mailer can't send.
   config.action_mailer.raise_delivery_errors = false
 
+  config.action_mailer.perform_caching = false
+
   # Print deprecation notices to the Rails logger.
   config.active_support.deprecation = :log
 
@@ -27,15 +42,13 @@
   # number of complex assets.
   config.assets.debug = true
 
-  # Asset digests allow you to set far-future HTTP expiration dates on all assets,
-  # yet still be able to expire them through the digest params.
-  config.assets.digest = true
-
-  # Adds additional error checking when serving assets at runtime.
-  # Checks for improperly declared sprockets dependencies.
-  # Raises helpful error messages.
-  config.assets.raise_runtime_errors = true
+  # Suppress logger output for asset requests.
+  config.assets.quiet = true
 
   # Raises error for missing translations
   # config.action_view.raise_on_missing_translations = true
+
+  # Use an evented file watcher to asynchronously detect changes in source code,
+  # routes, locales, etc. This feature depends on the listen gem.
+  config.file_watcher = ActiveSupport::EventedFileUpdateChecker
 end
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/config/environments/production.rb rails 5.0/config/environments/production.rb
--- rails 4.2/config/environments/production.rb	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/config/environments/production.rb	2022-10-13 09:23:33.000000000 -0700
@@ -14,15 +14,9 @@
   config.consider_all_requests_local       = false
   config.action_controller.perform_caching = true
 
-  # Enable Rack::Cache to put a simple HTTP cache in front of your application
-  # Add `rack-cache` to your Gemfile before enabling this.
-  # For large-scale production use, consider using a caching reverse proxy like
-  # NGINX, varnish or squid.
-  # config.action_dispatch.rack_cache = true
-
   # Disable serving static files from the `/public` folder by default since
   # Apache or NGINX already handles this.
-  config.serve_static_files = ENV['RAILS_SERVE_STATIC_FILES'].present?
+  config.public_file_server.enabled = ENV['RAILS_SERVE_STATIC_FILES'].present?
 
   # Compress JavaScripts and CSS.
   config.assets.js_compressor = :uglifier
@@ -31,16 +25,20 @@
   # Do not fallback to assets pipeline if a precompiled asset is missed.
   config.assets.compile = false
 
-  # Asset digests allow you to set far-future HTTP expiration dates on all assets,
-  # yet still be able to expire them through the digest params.
-  config.assets.digest = true
-
   # `config.assets.precompile` and `config.assets.version` have moved to config/initializers/assets.rb
 
+  # Enable serving of images, stylesheets, and JavaScripts from an asset server.
+  # config.action_controller.asset_host = 'http://assets.example.com'
+
   # Specifies the header that your server uses for sending files.
   # config.action_dispatch.x_sendfile_header = 'X-Sendfile' # for Apache
   # config.action_dispatch.x_sendfile_header = 'X-Accel-Redirect' # for NGINX
 
+  # Mount Action Cable outside main process or domain
+  # config.action_cable.mount_path = nil
+  # config.action_cable.url = 'wss://example.com/cable'
+  # config.action_cable.allowed_request_origins = [ 'http://example.com', /http:\/\/example.*/ ]
+
   # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
   # config.force_ssl = true
 
@@ -49,16 +47,15 @@
   config.log_level = :debug
 
   # Prepend all log lines with the following tags.
-  # config.log_tags = [ :subdomain, :uuid ]
-
-  # Use a different logger for distributed setups.
-  # config.logger = ActiveSupport::TaggedLogging.new(SyslogLogger.new)
+  config.log_tags = [ :request_id ]
 
   # Use a different cache store in production.
   # config.cache_store = :mem_cache_store
 
-  # Enable serving of images, stylesheets, and JavaScripts from an asset server.
-  # config.action_controller.asset_host = 'http://assets.example.com'
+  # Use a real queuing backend for Active Job (and separate queues per environment)
+  # config.active_job.queue_adapter     = :resque
+  # config.active_job.queue_name_prefix = "r50_#{Rails.env}"
+  config.action_mailer.perform_caching = false
 
   # Ignore bad email addresses and do not raise email delivery errors.
   # Set this to true and configure the email server for immediate delivery to raise delivery errors.
@@ -74,6 +71,16 @@
   # Use default logging formatter so that PID and timestamp are not suppressed.
   config.log_formatter = ::Logger::Formatter.new
 
+  # Use a different logger for distributed setups.
+  # require 'syslog/logger'
+  # config.logger = ActiveSupport::TaggedLogging.new(Syslog::Logger.new 'app-name')
+
+  if ENV["RAILS_LOG_TO_STDOUT"].present?
+    logger           = ActiveSupport::Logger.new(STDOUT)
+    logger.formatter = config.log_formatter
+    config.logger = ActiveSupport::TaggedLogging.new(logger)
+  end
+
   # Do not dump schema after migrations.
   config.active_record.dump_schema_after_migration = false
 end
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/config/environments/test.rb rails 5.0/config/environments/test.rb
--- rails 4.2/config/environments/test.rb	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/config/environments/test.rb	2022-10-13 09:23:33.000000000 -0700
@@ -12,9 +12,11 @@
   # preloads Rails for running tests, you may have to set it to true.
   config.eager_load = false
 
-  # Configure static file server for tests with Cache-Control for performance.
-  config.serve_static_files   = true
-  config.static_cache_control = 'public, max-age=3600'
+  # Configure public file server for tests with Cache-Control for performance.
+  config.public_file_server.enabled = true
+  config.public_file_server.headers = {
+    'Cache-Control' => 'public, max-age=3600'
+  }
 
   # Show full error reports and disable caching.
   config.consider_all_requests_local       = true
@@ -25,15 +27,13 @@
 
   # Disable request forgery protection in test environment.
   config.action_controller.allow_forgery_protection = false
+  config.action_mailer.perform_caching = false
 
   # Tell Action Mailer not to deliver emails to the real world.
   # The :test delivery method accumulates sent emails in the
   # ActionMailer::Base.deliveries array.
   config.action_mailer.delivery_method = :test
 
-  # Randomize the order test cases are executed.
-  config.active_support.test_order = :random
-
   # Print deprecation notices to the stderr.
   config.active_support.deprecation = :stderr
 
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/config/initializers/application_controller_renderer.rb rails 5.0/config/initializers/application_controller_renderer.rb
--- rails 4.2/config/initializers/application_controller_renderer.rb	1969-12-31 16:00:00.000000000 -0800
+++ rails 5.0/config/initializers/application_controller_renderer.rb	2022-10-13 09:23:33.000000000 -0700
@@ -0,0 +1,8 @@
+# Be sure to restart your server when you modify this file.
+
+# ActiveSupport::Reloader.to_prepare do
+#   ApplicationController.renderer.defaults.merge!(
+#     http_host: 'example.org',
+#     https: false
+#   )
+# end
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/config/initializers/cookies_serializer.rb rails 5.0/config/initializers/cookies_serializer.rb
--- rails 4.2/config/initializers/cookies_serializer.rb	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/config/initializers/cookies_serializer.rb	2022-10-13 09:23:33.000000000 -0700
@@ -1,3 +1,5 @@
 # Be sure to restart your server when you modify this file.
 
+# Specify a serializer for the signed and encrypted cookie jars.
+# Valid options are :json, :marshal, and :hybrid.
 Rails.application.config.action_dispatch.cookies_serializer = :json
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/config/initializers/new_framework_defaults.rb rails 5.0/config/initializers/new_framework_defaults.rb
--- rails 4.2/config/initializers/new_framework_defaults.rb	1969-12-31 16:00:00.000000000 -0800
+++ rails 5.0/config/initializers/new_framework_defaults.rb	2022-10-13 09:23:33.000000000 -0700
@@ -0,0 +1,26 @@
+# Be sure to restart your server when you modify this file.
+#
+# This file contains migration options to ease your Rails 5.0 upgrade.
+#
+# Read the Guide for Upgrading Ruby on Rails for more info on each option.
+
+Rails.application.config.action_controller.raise_on_unfiltered_parameters = true
+
+# Enable per-form CSRF tokens. Previous versions had false.
+Rails.application.config.action_controller.per_form_csrf_tokens = true
+
+# Enable origin-checking CSRF mitigation. Previous versions had false.
+Rails.application.config.action_controller.forgery_protection_origin_check = true
+
+# Make Ruby 2.4 preserve the timezone of the receiver when calling `to_time`.
+# Previous versions had false.
+ActiveSupport.to_time_preserves_timezone = true
+
+# Require `belongs_to` associations by default. Previous versions had false.
+Rails.application.config.active_record.belongs_to_required_by_default = true
+
+# Do not halt callback chains when a callback returns false. Previous versions had true.
+ActiveSupport.halt_callback_chains_on_return_false = false
+
+# Configure SSL options to enable HSTS with subdomains. Previous versions had false.
+Rails.application.config.ssl_options = { hsts: { subdomains: true } }
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/config/initializers/session_store.rb rails 5.0/config/initializers/session_store.rb
--- rails 4.2/config/initializers/session_store.rb	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/config/initializers/session_store.rb	2022-10-13 09:23:33.000000000 -0700
@@ -1,3 +1,3 @@
 # Be sure to restart your server when you modify this file.
 
-Rails.application.config.session_store :cookie_store, key: '_r42_session'
+Rails.application.config.session_store :cookie_store, key: '_r50_session'
Only in rails 4.2/config/initializers: to_time_preserves_timezone.rb
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/config/initializers/wrap_parameters.rb rails 5.0/config/initializers/wrap_parameters.rb
--- rails 4.2/config/initializers/wrap_parameters.rb	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/config/initializers/wrap_parameters.rb	2022-10-13 09:23:33.000000000 -0700
@@ -5,7 +5,7 @@
 
 # Enable parameter wrapping for JSON. You can disable this by setting :format to an empty array.
 ActiveSupport.on_load(:action_controller) do
-  wrap_parameters format: [:json] if respond_to?(:wrap_parameters)
+  wrap_parameters format: [:json]
 end
 
 # To enable root element in JSON for ActiveRecord objects.
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/config/puma.rb rails 5.0/config/puma.rb
--- rails 4.2/config/puma.rb	1969-12-31 16:00:00.000000000 -0800
+++ rails 5.0/config/puma.rb	2022-10-13 09:23:33.000000000 -0700
@@ -0,0 +1,47 @@
+# Puma can serve each request in a thread from an internal thread pool.
+# The `threads` method setting takes two numbers a minimum and maximum.
+# Any libraries that use thread pools should be configured to match
+# the maximum value specified for Puma. Default is set to 5 threads for minimum
+# and maximum, this matches the default thread size of Active Record.
+#
+threads_count = ENV.fetch("RAILS_MAX_THREADS") { 5 }.to_i
+threads threads_count, threads_count
+
+# Specifies the `port` that Puma will listen on to receive requests, default is 3000.
+#
+port        ENV.fetch("PORT") { 3000 }
+
+# Specifies the `environment` that Puma will run in.
+#
+environment ENV.fetch("RAILS_ENV") { "development" }
+
+# Specifies the number of `workers` to boot in clustered mode.
+# Workers are forked webserver processes. If using threads and workers together
+# the concurrency of the application would be max `threads` * `workers`.
+# Workers do not work on JRuby or Windows (both of which do not support
+# processes).
+#
+# workers ENV.fetch("WEB_CONCURRENCY") { 2 }
+
+# Use the `preload_app!` method when specifying a `workers` number.
+# This directive tells Puma to first boot the application and load code
+# before forking the application. This takes advantage of Copy On Write
+# process behavior so workers use less memory. If you use this option
+# you need to make sure to reconnect any threads in the `on_worker_boot`
+# block.
+#
+# preload_app!
+
+# The code in the `on_worker_boot` will be called if you are using
+# clustered mode by specifying a number of `workers`. After each worker
+# process is booted this block will be run, if you are using `preload_app!`
+# option you will want to use this block to reconnect to any threads
+# or connections that may have been created at application boot, Ruby
+# cannot share connections between processes.
+#
+# on_worker_boot do
+#   ActiveRecord::Base.establish_connection if defined?(ActiveRecord)
+# end
+
+# Allow puma to be restarted by `rails restart` command.
+plugin :tmp_restart
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/config/routes.rb rails 5.0/config/routes.rb
--- rails 4.2/config/routes.rb	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/config/routes.rb	2022-10-13 09:23:33.000000000 -0700
@@ -1,56 +1,3 @@
 Rails.application.routes.draw do
-  # The priority is based upon order of creation: first created -> highest priority.
-  # See how all your routes lay out with "rake routes".
-
-  # You can have the root of your site routed with "root"
-  # root 'welcome#index'
-
-  # Example of regular route:
-  #   get 'products/:id' => 'catalog#view'
-
-  # Example of named route that can be invoked with purchase_url(id: product.id)
-  #   get 'products/:id/purchase' => 'catalog#purchase', as: :purchase
-
-  # Example resource route (maps HTTP verbs to controller actions automatically):
-  #   resources :products
-
-  # Example resource route with options:
-  #   resources :products do
-  #     member do
-  #       get 'short'
-  #       post 'toggle'
-  #     end
-  #
-  #     collection do
-  #       get 'sold'
-  #     end
-  #   end
-
-  # Example resource route with sub-resources:
-  #   resources :products do
-  #     resources :comments, :sales
-  #     resource :seller
-  #   end
-
-  # Example resource route with more complex sub-resources:
-  #   resources :products do
-  #     resources :comments
-  #     resources :sales do
-  #       get 'recent', on: :collection
-  #     end
-  #   end
-
-  # Example resource route with concerns:
-  #   concern :toggleable do
-  #     post 'toggle'
-  #   end
-  #   resources :posts, concerns: :toggleable
-  #   resources :photos, concerns: :toggleable
-
-  # Example resource route within a namespace:
-  #   namespace :admin do
-  #     # Directs /admin/products/* to Admin::ProductsController
-  #     # (app/controllers/admin/products_controller.rb)
-  #     resources :products
-  #   end
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
 end
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/config/secrets.yml rails 5.0/config/secrets.yml
--- rails 4.2/config/secrets.yml	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/config/secrets.yml	2022-10-13 09:23:33.000000000 -0700
@@ -5,16 +5,16 @@
 
 # Make sure the secret is at least 30 characters and all random,
 # no regular words or you'll be exposed to dictionary attacks.
-# You can use `rake secret` to generate a secure secret key.
+# You can use `rails secret` to generate a secure secret key.
 
 # Make sure the secrets in this file are kept private
 # if you're sharing your code publicly.
 
 development:
-  secret_key_base: 10593f4fa2f3f8b30037b201cfb1cf9a5f1365e0d43a7d4e33945c0a2fbdbb7871a44c851aec45bd1c9c3e3144f16401fbc40f32289d627c4963498d3f82b147
+  secret_key_base: 5f9d61c6464d3cb7e58bad58ed55f3584ebb61bb93642e1c0279be0985f2290e49806be5fe7025e29ef4f0d4ce0485b89416d0043aa4f0f93ebf9e4428202495
 
 test:
-  secret_key_base: 364bb3705f8b6b72a09626305ce2b1557d8b9c135e7feefeb99dc3b419d72fefac94a34acafeb944c7ae5941005b2598fab2b137a2b02e4670f878a30d661f11
+  secret_key_base: 97a9e34311dc6ac988c185091a59cd1f598ab803eda85e19e547cfe5c394180e6c46c742066b95195850e4d3c58f40e57f9efd411dbdd349072144f8320dc9b7
 
 # Do not keep production secrets in the repository,
 # instead read values from the environment.
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/config/spring.rb rails 5.0/config/spring.rb
--- rails 4.2/config/spring.rb	1969-12-31 16:00:00.000000000 -0800
+++ rails 5.0/config/spring.rb	2022-10-13 09:23:33.000000000 -0700
@@ -0,0 +1,6 @@
+%w(
+  .ruby-version
+  .rbenv-vars
+  tmp/restart.txt
+  tmp/caching-dev.txt
+).each { |path| Spring.watch(path) }
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/config.ru rails 5.0/config.ru
--- rails 4.2/config.ru	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/config.ru	2022-10-13 09:23:33.000000000 -0700
@@ -1,4 +1,5 @@
 # This file is used by Rack-based servers to start the application.
 
-require ::File.expand_path('../config/environment', __FILE__)
+require_relative 'config/environment'
+
 run Rails.application
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/db/seeds.rb rails 5.0/db/seeds.rb
--- rails 4.2/db/seeds.rb	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/db/seeds.rb	2022-10-13 09:23:33.000000000 -0700
@@ -1,7 +1,7 @@
 # This file should contain all the record creation needed to seed the database with its default values.
-# The data can then be loaded with the rake db:seed (or created alongside the db with db:setup).
+# The data can then be loaded with the rails db:seed command (or created alongside the database with db:setup).
 #
 # Examples:
 #
-#   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
-#   Mayor.create(name: 'Emanuel', city: cities.first)
+#   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
+#   Character.create(name: 'Luke', movie: movies.first)
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/public/404.html rails 5.0/public/404.html
--- rails 4.2/public/404.html	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/public/404.html	2022-10-13 09:23:33.000000000 -0700
@@ -4,7 +4,7 @@
   <title>The page you were looking for doesn't exist (404)</title>
   <meta name="viewport" content="width=device-width,initial-scale=1">
   <style>
-  body {
+  .rails-default-error-page {
     background-color: #EFEFEF;
     color: #2E2F30;
     text-align: center;
@@ -12,13 +12,13 @@
     margin: 0;
   }
 
-  div.dialog {
+  .rails-default-error-page div.dialog {
     width: 95%;
     max-width: 33em;
     margin: 4em auto 0;
   }
 
-  div.dialog > div {
+  .rails-default-error-page div.dialog > div {
     border: 1px solid #CCC;
     border-right-color: #999;
     border-left-color: #999;
@@ -31,13 +31,13 @@
     box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
 
-  h1 {
+  .rails-default-error-page h1 {
     font-size: 100%;
     color: #730E15;
     line-height: 1.5em;
   }
 
-  div.dialog > p {
+  .rails-default-error-page div.dialog > p {
     margin: 0 0 1em;
     padding: 1em;
     background-color: #F7F7F7;
@@ -54,7 +54,7 @@
   </style>
 </head>
 
-<body>
+<body class="rails-default-error-page">
   <!-- This file lives in public/404.html -->
   <div class="dialog">
     <div>
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/public/422.html rails 5.0/public/422.html
--- rails 4.2/public/422.html	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/public/422.html	2022-10-13 09:23:33.000000000 -0700
@@ -4,7 +4,7 @@
   <title>The change you wanted was rejected (422)</title>
   <meta name="viewport" content="width=device-width,initial-scale=1">
   <style>
-  body {
+  .rails-default-error-page {
     background-color: #EFEFEF;
     color: #2E2F30;
     text-align: center;
@@ -12,13 +12,13 @@
     margin: 0;
   }
 
-  div.dialog {
+  .rails-default-error-page div.dialog {
     width: 95%;
     max-width: 33em;
     margin: 4em auto 0;
   }
 
-  div.dialog > div {
+  .rails-default-error-page div.dialog > div {
     border: 1px solid #CCC;
     border-right-color: #999;
     border-left-color: #999;
@@ -31,13 +31,13 @@
     box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
 
-  h1 {
+  .rails-default-error-page h1 {
     font-size: 100%;
     color: #730E15;
     line-height: 1.5em;
   }
 
-  div.dialog > p {
+  .rails-default-error-page div.dialog > p {
     margin: 0 0 1em;
     padding: 1em;
     background-color: #F7F7F7;
@@ -54,7 +54,7 @@
   </style>
 </head>
 
-<body>
+<body class="rails-default-error-page">
   <!-- This file lives in public/422.html -->
   <div class="dialog">
     <div>
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 4.2/public/500.html rails 5.0/public/500.html
--- rails 4.2/public/500.html	2022-10-13 15:13:04.000000000 -0700
+++ rails 5.0/public/500.html	2022-10-13 09:23:33.000000000 -0700
@@ -4,7 +4,7 @@
   <title>We're sorry, but something went wrong (500)</title>
   <meta name="viewport" content="width=device-width,initial-scale=1">
   <style>
-  body {
+  .rails-default-error-page {
     background-color: #EFEFEF;
     color: #2E2F30;
     text-align: center;
@@ -12,13 +12,13 @@
     margin: 0;
   }
 
-  div.dialog {
+  .rails-default-error-page div.dialog {
     width: 95%;
     max-width: 33em;
     margin: 4em auto 0;
   }
 
-  div.dialog > div {
+  .rails-default-error-page div.dialog > div {
     border: 1px solid #CCC;
     border-right-color: #999;
     border-left-color: #999;
@@ -31,13 +31,13 @@
     box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
   }
 
-  h1 {
+  .rails-default-error-page h1 {
     font-size: 100%;
     color: #730E15;
     line-height: 1.5em;
   }
 
-  div.dialog > p {
+  .rails-default-error-page div.dialog > p {
     margin: 0 0 1em;
     padding: 1em;
     background-color: #F7F7F7;
@@ -54,7 +54,7 @@
   </style>
 </head>
 
-<body>
+<body class="rails-default-error-page">
   <!-- This file lives in public/500.html -->
   <div class="dialog">
     <div>
