diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/.gitignore rails 5.1/.gitignore
--- rails 5.0/.gitignore	2025-01-10 11:16:48
+++ rails 5.1/.gitignore	2025-01-10 11:16:51
@@ -17,5 +17,7 @@
 !/log/.keep
 !/tmp/.keep
 
-# Ignore Byebug command history file.
+/node_modules
+/yarn-error.log
+
 .byebug_history
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/Gemfile rails 5.1/Gemfile
--- rails 5.0/Gemfile	2025-01-10 11:16:48
+++ rails 5.1/Gemfile	2025-01-10 11:16:51
@@ -7,28 +7,26 @@
 
 
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
-gem 'rails', '~> 5.0.7', '>= 5.0.7.2'
+gem 'rails', '~> 5.1.7'
 # Use sqlite3 as the database for Active Record
 gem 'sqlite3'
 # Use Puma as the app server
-gem 'puma', '~> 3.0'
+gem 'puma', '~> 3.7'
 # Use SCSS for stylesheets
 gem 'sass-rails', '~> 5.0'
 # Use Uglifier as compressor for JavaScript assets
 gem 'uglifier', '>= 1.3.0'
-# Use CoffeeScript for .coffee assets and views
-gem 'coffee-rails', '~> 4.2'
 # See https://github.com/rails/execjs#readme for more supported runtimes
 # gem 'therubyracer', platforms: :ruby
 
-# Use jquery as the JavaScript library
-gem 'jquery-rails'
+# Use CoffeeScript for .coffee assets and views
+gem 'coffee-rails', '~> 4.2'
 # Turbolinks makes navigating your web application faster. Read more: https://github.com/turbolinks/turbolinks
 gem 'turbolinks', '~> 5'
 # Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
 gem 'jbuilder', '~> 2.5'
 # Use Redis adapter to run Action Cable in production
-# gem 'redis', '~> 3.0'
+# gem 'redis', '~> 4.0'
 # Use ActiveModel has_secure_password
 # gem 'bcrypt', '~> 3.1.7'
 
@@ -37,13 +35,16 @@
 
 group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
-  gem 'byebug', platform: :mri
+  gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
+  # Adds support for Capybara system testing and selenium driver
+  gem 'capybara', '>= 2.15'
+  gem 'selenium-webdriver'
 end
 
 group :development do
   # Access an IRB console on exception pages or by using <%= console %> anywhere in the code.
   gem 'web-console', '>= 3.3.0'
-  gem 'listen', '~> 3.0.5'
+  gem 'listen', '>= 3.0.5', '< 3.2'
   # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
   gem 'spring'
   gem 'spring-watcher-listen', '~> 2.0.0'
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/Gemfile.lock rails 5.1/Gemfile.lock
--- rails 5.0/Gemfile.lock	2025-01-10 11:16:50
+++ rails 5.1/Gemfile.lock	2025-01-10 11:16:53
@@ -1,48 +1,59 @@
 GEM
   remote: https://rubygems.org/
   specs:
-    actioncable (5.0.7.2)
-      actionpack (= 5.0.7.2)
-      nio4r (>= 1.2, < 3.0)
+    actioncable (5.1.7)
+      actionpack (= 5.1.7)
+      nio4r (~> 2.0)
       websocket-driver (~> 0.6.1)
-    actionmailer (5.0.7.2)
-      actionpack (= 5.0.7.2)
-      actionview (= 5.0.7.2)
-      activejob (= 5.0.7.2)
+    actionmailer (5.1.7)
+      actionpack (= 5.1.7)
+      actionview (= 5.1.7)
+      activejob (= 5.1.7)
       mail (~> 2.5, >= 2.5.4)
       rails-dom-testing (~> 2.0)
-    actionpack (5.0.7.2)
-      actionview (= 5.0.7.2)
-      activesupport (= 5.0.7.2)
+    actionpack (5.1.7)
+      actionview (= 5.1.7)
+      activesupport (= 5.1.7)
       rack (~> 2.0)
-      rack-test (~> 0.6.3)
+      rack-test (>= 0.6.3)
       rails-dom-testing (~> 2.0)
       rails-html-sanitizer (~> 1.0, >= 1.0.2)
-    actionview (5.0.7.2)
-      activesupport (= 5.0.7.2)
+    actionview (5.1.7)
+      activesupport (= 5.1.7)
       builder (~> 3.1)
-      erubis (~> 2.7.0)
+      erubi (~> 1.4)
       rails-dom-testing (~> 2.0)
       rails-html-sanitizer (~> 1.0, >= 1.0.3)
-    activejob (5.0.7.2)
-      activesupport (= 5.0.7.2)
+    activejob (5.1.7)
+      activesupport (= 5.1.7)
       globalid (>= 0.3.6)
-    activemodel (5.0.7.2)
-      activesupport (= 5.0.7.2)
-    activerecord (5.0.7.2)
-      activemodel (= 5.0.7.2)
-      activesupport (= 5.0.7.2)
-      arel (~> 7.0)
-    activesupport (5.0.7.2)
+    activemodel (5.1.7)
+      activesupport (= 5.1.7)
+    activerecord (5.1.7)
+      activemodel (= 5.1.7)
+      activesupport (= 5.1.7)
+      arel (~> 8.0)
+    activesupport (5.1.7)
       concurrent-ruby (~> 1.0, >= 1.0.2)
       i18n (>= 0.7, < 2)
       minitest (~> 5.1)
       tzinfo (~> 1.1)
-    arel (7.1.4)
+    addressable (2.8.7)
+      public_suffix (>= 2.0.2, < 7.0)
+    arel (8.0.0)
     base64 (0.2.0)
     bindex (0.8.1)
     builder (3.3.0)
     byebug (11.1.3)
+    capybara (3.39.2)
+      addressable
+      matrix
+      mini_mime (>= 0.1.3)
+      nokogiri (~> 1.8)
+      rack (>= 1.6.0)
+      rack-test (>= 0.6.3)
+      regexp_parser (>= 1.5, < 3.0)
+      xpath (~> 3.2)
     coffee-rails (4.2.2)
       coffee-script (>= 2.2.0)
       railties (>= 4.0.0)
@@ -53,7 +64,7 @@
     concurrent-ruby (1.3.4)
     crass (1.0.6)
     date (3.4.1)
-    erubis (2.7.0)
+    erubi (1.13.1)
     execjs (2.10.0)
     ffi (1.17.1-arm64-darwin)
     globalid (1.1.0)
@@ -63,13 +74,10 @@
     jbuilder (2.13.0)
       actionview (>= 5.0.0)
       activesupport (>= 5.0.0)
-    jquery-rails (4.6.0)
-      rails-dom-testing (>= 1, < 3)
-      railties (>= 4.2.0)
-      thor (>= 0.14, < 2.0)
-    listen (3.0.8)
+    listen (3.1.5)
       rb-fsevent (~> 0.9, >= 0.9.4)
       rb-inotify (~> 0.9, >= 0.9.7)
+      ruby_dep (~> 1.2)
     loofah (2.24.0)
       crass (~> 1.0.2)
       nokogiri (>= 1.12.0)
@@ -78,6 +86,7 @@
       net-imap
       net-pop
       net-smtp
+    matrix (0.4.2)
     method_source (1.1.0)
     mini_mime (1.1.5)
     mini_portile2 (2.8.8)
@@ -94,22 +103,23 @@
     nio4r (2.7.4)
     nokogiri (1.15.7-arm64-darwin)
       racc (~> 1.4)
+    public_suffix (5.1.1)
     puma (3.12.6)
     racc (1.8.1)
     rack (2.2.10)
-    rack-test (0.6.3)
-      rack (>= 1.0)
-    rails (5.0.7.2)
-      actioncable (= 5.0.7.2)
-      actionmailer (= 5.0.7.2)
-      actionpack (= 5.0.7.2)
-      actionview (= 5.0.7.2)
-      activejob (= 5.0.7.2)
-      activemodel (= 5.0.7.2)
-      activerecord (= 5.0.7.2)
-      activesupport (= 5.0.7.2)
+    rack-test (2.2.0)
+      rack (>= 1.3)
+    rails (5.1.7)
+      actioncable (= 5.1.7)
+      actionmailer (= 5.1.7)
+      actionpack (= 5.1.7)
+      actionview (= 5.1.7)
+      activejob (= 5.1.7)
+      activemodel (= 5.1.7)
+      activerecord (= 5.1.7)
+      activesupport (= 5.1.7)
       bundler (>= 1.3.0)
-      railties (= 5.0.7.2)
+      railties (= 5.1.7)
       sprockets-rails (>= 2.0.0)
     rails-dom-testing (2.2.0)
       activesupport (>= 5.0.0)
@@ -118,9 +128,9 @@
     rails-html-sanitizer (1.6.2)
       loofah (~> 2.21)
       nokogiri (>= 1.15.7, != 1.16.7, != 1.16.6, != 1.16.5, != 1.16.4, != 1.16.3, != 1.16.2, != 1.16.1, != 1.16.0.rc1, != 1.16.0)
-    railties (5.0.7.2)
-      actionpack (= 5.0.7.2)
-      activesupport (= 5.0.7.2)
+    railties (5.1.7)
+      actionpack (= 5.1.7)
+      activesupport (= 5.1.7)
       method_source
       rake (>= 0.8.7)
       thor (>= 0.18.1, < 2.0)
@@ -128,6 +138,10 @@
     rb-fsevent (0.11.2)
     rb-inotify (0.11.1)
       ffi (~> 1.0)
+    regexp_parser (2.10.0)
+    rexml (3.4.0)
+    ruby_dep (1.5.0)
+    rubyzip (2.4.1)
     sass (3.7.4)
       sass-listen (~> 4.0.0)
     sass-listen (4.0.0)
@@ -139,6 +153,10 @@
       sprockets (>= 2.8, < 4.0)
       sprockets-rails (>= 2.0, < 4.0)
       tilt (>= 1.1, < 3)
+    selenium-webdriver (4.9.0)
+      rexml (~> 3.2, >= 3.2.5)
+      rubyzip (>= 1.2.2, < 3.0)
+      websocket (~> 1.0)
     spring (2.1.1)
     spring-watcher-listen (2.0.1)
       listen (>= 2.7, < 4.0)
@@ -169,22 +187,26 @@
       activemodel (>= 5.0)
       bindex (>= 0.4.0)
       railties (>= 5.0)
+    websocket (1.2.11)
     websocket-driver (0.6.5)
       websocket-extensions (>= 0.1.0)
     websocket-extensions (0.1.5)
+    xpath (3.2.0)
+      nokogiri (~> 1.8)
 
 PLATFORMS
   arm64-darwin-22
 
 DEPENDENCIES
   byebug
+  capybara (>= 2.15)
   coffee-rails (~> 4.2)
   jbuilder (~> 2.5)
-  jquery-rails
-  listen (~> 3.0.5)
-  puma (~> 3.0)
-  rails (~> 5.0.7, >= 5.0.7.2)
+  listen (>= 3.0.5, < 3.2)
+  puma (~> 3.7)
+  rails (~> 5.1.7)
   sass-rails (~> 5.0)
+  selenium-webdriver
   spring
   spring-watcher-listen (~> 2.0.0)
   sqlite3
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/app/assets/javascripts/application.js rails 5.1/app/assets/javascripts/application.js
--- rails 5.0/app/assets/javascripts/application.js	2025-01-10 11:16:48
+++ rails 5.1/app/assets/javascripts/application.js	2025-01-10 11:16:51
@@ -1,8 +1,8 @@
 // This is a manifest file that'll be compiled into application.js, which will include all the files
 // listed below.
 //
-// Any JavaScript/Coffee file within this directory, lib/assets/javascripts, vendor/assets/javascripts,
-// or any plugin's vendor/assets/javascripts directory can be referenced here using a relative path.
+// Any JavaScript/Coffee file within this directory, lib/assets/javascripts, or any plugin's
+// vendor/assets/javascripts directory can be referenced here using a relative path.
 //
 // It's not advisable to add code directly here, but if you do, it'll appear at the bottom of the
 // compiled file. JavaScript code in this file should be added after the last require_* statement.
@@ -10,7 +10,6 @@
 // Read Sprockets README (https://github.com/rails/sprockets#sprockets-directives) for details
 // about supported directives.
 //
-//= require jquery
-//= require jquery_ujs
+//= require rails-ujs
 //= require turbolinks
 //= require_tree .
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/app/assets/javascripts/cable.js rails 5.1/app/assets/javascripts/cable.js
--- rails 5.0/app/assets/javascripts/cable.js	2025-01-10 11:16:48
+++ rails 5.1/app/assets/javascripts/cable.js	2025-01-10 11:16:51
@@ -1,5 +1,5 @@
 // Action Cable provides the framework to deal with WebSockets in Rails.
-// You can generate new channels where WebSocket features live using the rails generate channel command.
+// You can generate new channels where WebSocket features live using the `rails generate channel` command.
 //
 //= require action_cable
 //= require_self
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/app/assets/stylesheets/application.css rails 5.1/app/assets/stylesheets/application.css
--- rails 5.0/app/assets/stylesheets/application.css	2025-01-10 11:16:48
+++ rails 5.1/app/assets/stylesheets/application.css	2025-01-10 11:16:51
@@ -2,8 +2,8 @@
  * This is a manifest file that'll be compiled into application.css, which will include all the files
  * listed below.
  *
- * Any CSS and SCSS file within this directory, lib/assets/stylesheets, vendor/assets/stylesheets,
- * or any plugin's vendor/assets/stylesheets directory can be referenced here using a relative path.
+ * Any CSS and SCSS file within this directory, lib/assets/stylesheets, or any plugin's
+ * vendor/assets/stylesheets directory can be referenced here using a relative path.
  *
  * You're free to add application-wide styles to this file and they'll appear at the bottom of the
  * compiled file so the styles you add here take precedence over styles defined in any other CSS/SCSS
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/app/views/layouts/application.html.erb rails 5.1/app/views/layouts/application.html.erb
--- rails 5.0/app/views/layouts/application.html.erb	2025-01-10 11:16:48
+++ rails 5.1/app/views/layouts/application.html.erb	2025-01-10 11:16:51
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title>Rails50</title>
+    <title>Rails51</title>
     <%= csrf_meta_tags %>
 
     <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/bin/setup rails 5.1/bin/setup
--- rails 5.0/bin/setup	2025-01-10 11:16:48
+++ rails 5.1/bin/setup	2025-01-10 11:16:51
@@ -18,6 +18,10 @@
   system! 'gem install bundler --conservative'
   system('bundle check') || system!('bundle install')
 
+  # Install JavaScript dependencies if using Yarn
+  # system('bin/yarn')
+
+
   # puts "\n== Copying sample files =="
   # unless File.exist?('config/database.yml')
   #   cp 'config/database.yml.sample', 'config/database.yml'
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/bin/yarn rails 5.1/bin/yarn
--- rails 5.0/bin/yarn	1969-12-31 16:00:00
+++ rails 5.1/bin/yarn	2025-01-10 11:16:51
@@ -0,0 +1,11 @@
+#!/usr/bin/env ruby
+VENDOR_PATH = File.expand_path('..', __dir__)
+Dir.chdir(VENDOR_PATH) do
+  begin
+    exec "yarnpkg #{ARGV.join(" ")}"
+  rescue Errno::ENOENT
+    $stderr.puts "Yarn executable was not detected in the system."
+    $stderr.puts "Download Yarn at https://yarnpkg.com/en/docs/install"
+    exit 1
+  end
+end
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/config/application.rb rails 5.1/config/application.rb
--- rails 5.0/config/application.rb	2025-01-10 11:16:48
+++ rails 5.1/config/application.rb	2025-01-10 11:16:51
@@ -6,8 +6,11 @@
 # you've limited to :test, :development, or :production.
 Bundler.require(*Rails.groups)
 
-module Rails50
+module Rails51
   class Application < Rails::Application
+    # Initialize configuration defaults for originally generated Rails version.
+    config.load_defaults 5.1
+
     # Settings in config/environments/* take precedence over those specified here.
     # Application configuration should go into files in config/initializers
     # -- all .rb files in that directory are automatically loaded.
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/config/cable.yml rails 5.1/config/cable.yml
--- rails 5.0/config/cable.yml	2025-01-10 11:16:48
+++ rails 5.1/config/cable.yml	2025-01-10 11:16:51
@@ -7,3 +7,4 @@
 production:
   adapter: redis
   url: redis://localhost:6379/1
+  channel_prefix: rails_5_1_production
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/config/database.yml rails 5.1/config/database.yml
--- rails 5.0/config/database.yml	2025-01-10 11:16:48
+++ rails 5.1/config/database.yml	2025-01-10 11:16:51
@@ -6,7 +6,7 @@
 #
 default: &default
   adapter: sqlite3
-  pool: 5
+  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
   timeout: 5000
 
 development:
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/config/environments/development.rb rails 5.1/config/environments/development.rb
--- rails 5.0/config/environments/development.rb	2025-01-10 11:16:48
+++ rails 5.1/config/environments/development.rb	2025-01-10 11:16:51
@@ -18,7 +18,7 @@
 
     config.cache_store = :memory_store
     config.public_file_server.headers = {
-      'Cache-Control' => 'public, max-age=172800'
+      'Cache-Control' => "public, max-age=#{2.days.seconds.to_i}"
     }
   else
     config.action_controller.perform_caching = false
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/config/environments/production.rb rails 5.1/config/environments/production.rb
--- rails 5.0/config/environments/production.rb	2025-01-10 11:16:48
+++ rails 5.1/config/environments/production.rb	2025-01-10 11:16:51
@@ -14,6 +14,11 @@
   config.consider_all_requests_local       = false
   config.action_controller.perform_caching = true
 
+  # Attempt to read encrypted secrets from `config/secrets.yml.enc`.
+  # Requires an encryption key in `ENV["RAILS_MASTER_KEY"]` or
+  # `config/secrets.yml.key`.
+  config.read_encrypted_secrets = true
+
   # Disable serving static files from the `/public` folder by default since
   # Apache or NGINX already handles this.
   config.public_file_server.enabled = ENV['RAILS_SERVE_STATIC_FILES'].present?
@@ -54,7 +59,7 @@
 
   # Use a real queuing backend for Active Job (and separate queues per environment)
   # config.active_job.queue_adapter     = :resque
-  # config.active_job.queue_name_prefix = "rails_5_0_#{Rails.env}"
+  # config.active_job.queue_name_prefix = "rails_5_1_#{Rails.env}"
   config.action_mailer.perform_caching = false
 
   # Ignore bad email addresses and do not raise email delivery errors.
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/config/environments/test.rb rails 5.1/config/environments/test.rb
--- rails 5.0/config/environments/test.rb	2025-01-10 11:16:48
+++ rails 5.1/config/environments/test.rb	2025-01-10 11:16:51
@@ -15,7 +15,7 @@
   # Configure public file server for tests with Cache-Control for performance.
   config.public_file_server.enabled = true
   config.public_file_server.headers = {
-    'Cache-Control' => 'public, max-age=3600'
+    'Cache-Control' => "public, max-age=#{1.hour.seconds.to_i}"
   }
 
   # Show full error reports and disable caching.
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/config/initializers/assets.rb rails 5.1/config/initializers/assets.rb
--- rails 5.0/config/initializers/assets.rb	2025-01-10 11:16:48
+++ rails 5.1/config/initializers/assets.rb	2025-01-10 11:16:51
@@ -3,9 +3,12 @@
 # Version of your assets, change this if you want to expire all your assets.
 Rails.application.config.assets.version = '1.0'
 
-# Add additional assets to the asset load path
+# Add additional assets to the asset load path.
 # Rails.application.config.assets.paths << Emoji.images_path
+# Add Yarn node_modules folder to the asset load path.
+Rails.application.config.assets.paths << Rails.root.join('node_modules')
 
 # Precompile additional assets.
-# application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
-# Rails.application.config.assets.precompile += %w( search.js )
+# application.js, application.css, and all non-JS/CSS in the app/assets
+# folder are already added.
+# Rails.application.config.assets.precompile += %w( admin.js admin.css )
Only in rails 5.0/config/initializers: new_framework_defaults.rb
Only in rails 5.0/config/initializers: session_store.rb
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/config/locales/en.yml rails 5.1/config/locales/en.yml
--- rails 5.0/config/locales/en.yml	2025-01-10 11:16:48
+++ rails 5.1/config/locales/en.yml	2025-01-10 11:16:51
@@ -16,6 +16,16 @@
 #
 # This would use the information in config/locales/es.yml.
 #
+# The following keys must be escaped otherwise they will not be retrieved by
+# the default I18n backend:
+#
+# true, false, on, off, yes, no
+#
+# Instead, surround them with single quotes.
+#
+# en:
+#   'true': 'foo'
+#
 # To learn more, please read the Rails Internationalization guide
 # available at http://guides.rubyonrails.org/i18n.html.
 
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/config/puma.rb rails 5.1/config/puma.rb
--- rails 5.0/config/puma.rb	2025-01-10 11:16:48
+++ rails 5.1/config/puma.rb	2025-01-10 11:16:51
@@ -1,13 +1,13 @@
 # Puma can serve each request in a thread from an internal thread pool.
-# The `threads` method setting takes two numbers a minimum and maximum.
+# The `threads` method setting takes two numbers: a minimum and maximum.
 # Any libraries that use thread pools should be configured to match
 # the maximum value specified for Puma. Default is set to 5 threads for minimum
-# and maximum, this matches the default thread size of Active Record.
+# and maximum; this matches the default thread size of Active Record.
 #
-threads_count = ENV.fetch("RAILS_MAX_THREADS") { 5 }.to_i
+threads_count = ENV.fetch("RAILS_MAX_THREADS") { 5 }
 threads threads_count, threads_count
 
-# Specifies the `port` that Puma will listen on to receive requests, default is 3000.
+# Specifies the `port` that Puma will listen on to receive requests; default is 3000.
 #
 port        ENV.fetch("PORT") { 3000 }
 
@@ -32,16 +32,25 @@
 #
 # preload_app!
 
+# If you are preloading your application and using Active Record, it's
+# recommended that you close any connections to the database before workers
+# are forked to prevent connection leakage.
+#
+# before_fork do
+#   ActiveRecord::Base.connection_pool.disconnect! if defined?(ActiveRecord)
+# end
+
 # The code in the `on_worker_boot` will be called if you are using
 # clustered mode by specifying a number of `workers`. After each worker
-# process is booted this block will be run, if you are using `preload_app!`
-# option you will want to use this block to reconnect to any threads
-# or connections that may have been created at application boot, Ruby
+# process is booted, this block will be run. If you are using the `preload_app!`
+# option, you will want to use this block to reconnect to any threads
+# or connections that may have been created at application boot, as Ruby
 # cannot share connections between processes.
 #
 # on_worker_boot do
 #   ActiveRecord::Base.establish_connection if defined?(ActiveRecord)
 # end
+#
 
 # Allow puma to be restarted by `rails restart` command.
 plugin :tmp_restart
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/config/secrets.yml rails 5.1/config/secrets.yml
--- rails 5.0/config/secrets.yml	2025-01-10 11:16:48
+++ rails 5.1/config/secrets.yml	2025-01-10 11:16:51
@@ -10,13 +10,23 @@
 # Make sure the secrets in this file are kept private
 # if you're sharing your code publicly.
 
+# Shared secrets are available across all environments.
+
+# shared:
+#   api_key: a1B2c3D4e5F6
+
+# Environmental secrets are only available for that specific environment.
+
 development:
-  secret_key_base: e6a7a4605c7c59c67896fc80f2c4d0fbff8cd574f1fadce5cdc68f45d48e3449474d6ab5a226134b23a8c0030e44cfae455af1058c2fc60686ab7977032605ee
+  secret_key_base: 8e6837f3a8a418b80a489f41e17adc105a05aa095f557a849dbef94968be5ff511d761bbb03e9b673f75a7b79a784c7b7076024ccef3f8f6514b2fb668156aaa
 
 test:
-  secret_key_base: 05da054823427d33620f4987de8c34cd6cb2080ca216d6beea162b3dd3411c679057ea2867ab9fa1145e07830f9241bdc6e243031073545a575497788d0d71c7
+  secret_key_base: 40c12c8285d81d7455756ab9e16e035437b11d196bbb1414f190245b849fcb4571dbacbeb5906dc668e02ee90d0e1590e2d8956d0e34d73a7f672f75ea05b659
 
-# Do not keep production secrets in the repository,
-# instead read values from the environment.
+# Do not keep production secrets in the unencrypted secrets file.
+# Instead, either read values from the environment.
+# Or, use `bin/rails secrets:setup` to configure encrypted secrets
+# and move the `production:` environment over there.
+
 production:
   secret_key_base: <%= ENV["SECRET_KEY_BASE"] %>
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/package.json rails 5.1/package.json
--- rails 5.0/package.json	1969-12-31 16:00:00
+++ rails 5.1/package.json	2025-01-10 11:16:51
@@ -0,0 +1,5 @@
+{
+  "name": "rails_5_1",
+  "private": true,
+  "dependencies": {}
+}
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/public/robots.txt rails 5.1/public/robots.txt
--- rails 5.0/public/robots.txt	2025-01-10 11:16:48
+++ rails 5.1/public/robots.txt	2025-01-10 11:16:51
@@ -1,5 +1 @@
 # See http://www.robotstxt.org/robotstxt.html for documentation on how to use the robots.txt file
-#
-# To ban all spiders from the entire site uncomment the next two lines:
-# User-agent: *
-# Disallow: /
diff --ignore-space-change --minimal --recursive --unidirectional-new-file --unified rails 5.0/test/application_system_test_case.rb rails 5.1/test/application_system_test_case.rb
--- rails 5.0/test/application_system_test_case.rb	1969-12-31 16:00:00
+++ rails 5.1/test/application_system_test_case.rb	2025-01-10 11:16:51
@@ -0,0 +1,5 @@
+require "test_helper"
+
+class ApplicationSystemTestCase < ActionDispatch::SystemTestCase
+  driven_by :selenium, using: :chrome, screen_size: [1400, 1400]
+end
Only in rails 5.0/vendor: assets
